CI: ciliumidentity: TestUpdateUsedCIDIsReverted failed: lbmap mismatch

## CI failure

Discovered in PR https://github.com/cilium/cilium/pull/36602, but that PR is unrelated to the change here.

https://github.com/cilium/cilium/actions/runs/12795392193/job/35672703492#step:17:14138

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃   FAIL  package: github.com/cilium/cilium/operator/pkg/ciliumidentity   ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛

--- FAIL: TestUpdateUsedCIDIsReverted (0.64s)

    logger.go:257: time=2025-01-15T19:31:37.424Z level=INFO source=/host/operator/pkg/ciliumidentity/controller.go:159 msg="CID controller resource work queue configuration" workQueueSyncBackOff=1s workQueueMaxSyncBackOff=1m40s
    logger.go:257: time=2025-01-15T19:31:37.424Z level=INFO source=/host/vendor/github.com/cilium/hive/hive.go:338 msg="Starting hive"
    logger.go:257: time=2025-01-15T19:31:37.424Z level=INFO source=/host/operator/pkg/ciliumidentity/controller.go:122 msg="Starting CID controller Operator"
    logger.go:257: time=2025-01-15T19:31:37.424Z level=INFO source=/host/vendor/github.com/cilium/hive/hive.go:342 msg=Started duration=61.234µs
    logger.go:257: time=2025-01-15T19:31:37.424Z level=INFO source=/host/operator/pkg/ciliumidentity/reconciler.go:67 msg="Creating CID controller Operator reconciler"
    logger.go:257: time=2025-01-15T19:31:38.046Z level=INFO source=/host/operator/pkg/ciliumidentity/controller.go:211 msg="Starting CID controller reconciler"
    logger.go:257: time=2025-01-15T19:31:38.047Z level=INFO source=/host/operator/pkg/ciliumidentity/reconciler.go:387 msg="Reconcile all pods in namespace" k8sNamespace=ns1
    logger.go:257: time=2025-01-15T19:31:38.047Z level=INFO source=/host/operator/pkg/ciliumidentity/controller.go:216 msg="Starting resource worker"
    logger.go:257: time=2025-01-15T19:31:38.047Z level=INFO source=/host/operator/pkg/ciliumidentity/reconciler.go:294 msg="CID allocated for pod" k8sPodName=ns1/pod1 ciliumIdentityName=30308 oldIdentity="" labels="map[io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name:ns1 io.cilium.k8s.policy.cluster: io.kubernetes.pod.namespace:ns1 key-c:val-3]"
    logger.go:257: time=2025-01-15T19:31:38.047Z level=INFO source=/host/operator/pkg/ciliumidentity/reconciler.go:187 msg="Creating CID" labels="map[k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name:ns1 k8s:io.cilium.k8s.policy.cluster: k8s:io.kubernetes.pod.namespace:ns1 k8s:key-c:val-3]" ciliumIdentityName=30308
    logger.go:257: time=2025-01-15T19:31:38.056Z level=INFO source=/host/operator/pkg/ciliumidentity/reconciler.go:202 msg="Updating CID" ciliumIdentityName=30308
    controller_test.go:426: expected labels map[k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name:ns1 k8s:io.cilium.k8s.policy.cluster: k8s:io.kubernetes.pod.namespace:ns1 k8s:key-b:val-2], got map[k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name:ns1 k8s:io.cilium.k8s.policy.cluster: k8s:io.kubernetes.pod.namespace:ns1 k8s:key-c:val-3]
    logger.go:257: time=2025-01-15T19:31:38.057Z level=INFO source=/host/vendor/github.com/cilium/hive/hive.go:354 msg=Stopping
    logger.go:257: time=2025-01-15T19:31:38.059Z level=INFO source=/host/operator/pkg/ciliumidentity/controller.go:227 msg="Stopping resource worker"
    logger.go:257: time=2025-01-15T19:31:38.059Z level=INFO source=/host/pkg/hive/health/provider.go:57 msg="health.job-module-status-metrics (rev=7)" module=health
    logger.go:257: time=2025-01-15T19:31:38.059Z level=INFO source=/host/pkg/hive/health/provider.go:57 msg="test.job-op-managing-resource-wq (rev=8)" module=health
    logger.go:257: time=2025-01-15T19:31:38.059Z level=INFO source=/host/pkg/hive/health/provider.go:57 msg="test.job-proc-ces-events (rev=9)" module=health
    logger.go:257: time=2025-01-15T19:31:38.059Z level=INFO source=/host/pkg/hive/health/provider.go:57 msg="test.job-proc-cid-events (rev=10)" module=health
    logger.go:257: time=2025-01-15T19:31:38.059Z level=INFO source=/host/pkg/hive/health/provider.go:57 msg="test.job-proc-ns-events (rev=12)" module=health
    logger.go:257: time=2025-01-15T19:31:38.059Z level=INFO source=/host/pkg/hive/health/provider.go:57 msg="test.job-proc-pod-events (rev=11)" module=health
```

```
    testcase.go:352: validation failed: lbmap mismatch:
        --- /host/test/controlplane/services/nodeport/v1.26/lbmap1_nodeport-worker.golden
        +++ <actual>
        @@ -1,28 +1,17 @@
        - - Services --------------------------------------------------------------------------------------
        -| ID |                                      Name |      Type |             Frontend | Backend IDs |
        -|----+-------------------------------------------+-----------+----------------------+-------------
        -|  0 |                        default/kubernetes | ClusterIP |    10.96.0.1:443/TCP |           1 |
        -|  1 |                                 test/echo | ClusterIP |  10.96.50.104:80/TCP |           0 |
        -|  2 |                           test/echo-local | ClusterIP |  10.96.155.18:80/TCP |           0 |
        -|  3 | test/echo-757d4cb97f-9gmf7/host-port/8080 |  HostPort |     0.0.0.0:8080/TCP |           0 |
        -|  4 | test/echo-757d4cb97f-9gmf7/host-port/8080 |  HostPort |    10.0.0.2:8080/TCP |           0 |
        -|  5 | test/echo-757d4cb97f-9gmf7/host-port/8080 |  HostPort |    10.0.0.3:8080/TCP |           0 |
        -|  6 |                                 test/echo |  NodePort |    0.0.0.0:30781/TCP |           0 |
        -|  7 |                                 test/echo |  NodePort |   10.0.0.2:30781/TCP |           0 |
        -|  8 |                                 test/echo |  NodePort |   10.0.0.3:30781/TCP |           0 |
        -|  9 |                           test/echo-local |  NodePort |    0.0.0.0:30867/TCP |           0 |
        -| 10 |                           test/echo-local |  NodePort |  0.0.0.0:30867/TCP/i |           0 |
        -| 11 |                           test/echo-local |  NodePort |   10.0.0.2:30867/TCP |           0 |
        -| 12 |                           test/echo-local |  NodePort | 10.0.0.2:30867/TCP/i |           0 |
        -| 13 |                           test/echo-local |  NodePort |   10.0.0.3:30867/TCP |           0 |
        -| 14 |                           test/echo-local |  NodePort | 10.0.0.3:30867/TCP/i |           0 |
        - -------------------------------------------------------------------------------------------------
        + - Services -----------------------------------------------------------------------------------
        +| ID |                                      Name |      Type |          Frontend | Backend IDs |
        +|----+-------------------------------------------+-----------+-------------------+-------------
        +|  0 |                        default/kubernetes | ClusterIP | 10.96.0.1:443/TCP |           1 |
        +|  1 | test/echo-757d4cb97f-9gmf7/host-port/8080 |  HostPort |  0.0.0.0:8080/TCP |           0 |
        +|  2 | test/echo-757d4cb97f-9gmf7/host-port/8080 |  HostPort | 10.0.0.2:8080/TCP |           0 |
        +|  3 | test/echo-757d4cb97f-9gmf7/host-port/8080 |  HostPort | 10.0.0.3:8080/TCP |           0 |
        + ----------------------------------------------------------------------------------------------
         
        - - Backends ------------------------------------------------------------------------
        -| ID |            L3n4Addr |  State |                               Linked Services |
        -|----+---------------------+--------+-----------------------------------------------
        -|  0 | 10.244.1.224:80/TCP | active | 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14 |
        -|  1 | 172.18.0.4:6443/TCP | active |                                             0 |
        - -----------------------------------------------------------------------------------
        + - Backends ------------------------------------------
        +| ID |            L3n4Addr |  State | Linked Services |
        +|----+---------------------+--------+-----------------
        +|  0 | 10.244.1.224:80/TCP | active |         1, 2, 3 |
        +|  1 | 172.18.0.4:6443/TCP | active |               0 |
        + -----------------------------------------------------
```

--- a/operator/pkg/ciliumidentity/controller_test.go
+++ b/operator/pkg/ciliumidentity/controller_test.go
@@ -397,15 +397,6 @@ func TestUpdateUsedCIDIsReverted(t *testing.T) {
 		t.Fatalf("timeout waiting for identities in store: %s", lastErr)
 	}
 
-	// Start listening to identities events but discard all events being replayed.
-	events := (*cidResource).Events(ctx)
-	for ev := range events {
-		ev.Done(nil)
-		if ev.Kind == resource.Sync {
-			break
-		}
-	}
-
 	// Update identity.
 	updated := toUpdate.DeepCopy()
 	updated.Labels = cid2.Labels
@@ -414,31 +405,27 @@ func TestUpdateUsedCIDIsReverted(t *testing.T) {
 		t.Fatalf("update CID: %v", err)
 	}
 
-	// Wait for update event to propagate.
-	ev := <-events
-	if ev.Kind != resource.Upsert {
-		t.Fatalf("expected upsert event, got %v", ev.Kind)
+	cid, err := fakeClient.CiliumV2().CiliumIdentities().Get(ctx, updated.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("get CID: %v", err)
 	}
-	if ev.Key.Name != toUpdate.Name {
-		t.Fatalf("expected upsert event for cid %s, got %v", toUpdate.Name, ev.Key.Name)
+	if !cmp.Equal(cid.SecurityLabels, updated.SecurityLabels) {
+		t.Fatalf("expected labels %v, got %v", updated.SecurityLabels, cid.SecurityLabels)
 	}
-	if !cmp.Equal(ev.Object.SecurityLabels, updated.SecurityLabels) {
-		t.Fatalf("expected labels %v, got %v", updated.SecurityLabels, ev.Object.SecurityLabels)
-	}
-	ev.Done(nil)
 
-	// Wait for reconciler to revert the change.
-	ev = <-events
-	if ev.Kind != resource.Upsert {
-		t.Fatalf("expected upsert event, got %v", ev.Kind)
-	}
-	if ev.Key.Name != toUpdate.Name {
-		t.Fatalf("expected upsert event for cid %s, got %v", toUpdate.Name, ev.Key.Name)
-	}
-	if !cmp.Equal(ev.Object.SecurityLabels, cid1.SecurityLabels) {
-		t.Fatalf("expected labels %v, got %v", cid1.SecurityLabels, ev.Object.SecurityLabels)
-	}
-	ev.Done(nil)
+	require.EventuallyWithT(t, func(ct *assert.CollectT) {
+		cids := store.List()
+		assert.Len(ct, cids, 1)
+		if len(cids) != 1 {
+			return
+		}
+		cid = cids[0]
+
+		if !cmp.Equal(cid.SecurityLabels, cid1.SecurityLabels) {
+			t.Fatalf("expected labels %v, got %v", cid.SecurityLabels, cid1.SecurityLabels)
+		}
+	}, WaitUntilTimeout, 100*time.Millisecond)
+
 }
 
 func TestDeleteUsedCIDIsRecreated(t *testing.T) {

